code/controllers/stonedmc/subsystem/job.dm

```diff
@@ -61,31 +61,30 @@ SUBSYSTEM_DEF(job)
 	return type_occupations[jobtype]
 
 
-/datum/controller/subsystem/job/proc/AssignRole(mob/new_player/player, rank, latejoin = FALSE)
-	JobDebug("Running AR, Player: [player], Rank: [rank], LJ: [latejoin]")
-	if(player?.mind && rank)
-		var/datum/job/job = GetJob(rank)
+/datum/controller/subsystem/job/proc/AssignRole(mob/new_player/player, datum/job/job, latejoin = FALSE)
+	JobDebug("Running AR, Player: [player], Rank: [job?.title], LJ: [latejoin]")
+	if(player?.mind)
```

Comment:
If we remove `rank`, we need a way to prevent abstract jobs (e.g `datum/job/civilian`) to be assigned.